code/game/gamemodes/meteor/meteors.dm

```diff
@@ -153,16 +153,17 @@
 /obj/effect/meteor/proc/ram_turf(var/turf/T)
 	//first bust whatever is in the turf
 	for(var/atom/A in T)
-		if(A != src)
-			A.ex_act(hitpwr)
+		if(A == src) // Don't hit ourselves.
+			continue
+		if(isturf(A)) // Don't hit floors. We'll deal with walls later.

```

Comment:
I'm fairly sure that turfs do not contain themselves or any other turfs, so this if condition will never be true.